closed/openssl.gmk

```diff
@@ -18,50 +18,65 @@
 endif
 
 include $(SPEC)
+include $(TOPDIR)/make/common/MakeBase.gmk
 
-# config command guess the operating system and Configure openssl accordignly.
-# However, on some platforms (like AIX, Windows) it is essential to configure 
-# with one of the platform list provided by command ./Configure LIST command
-#     For AIX PPC64, it is aix64-cc
-#     For Linux ppc64le, it is linux-ppc64le
-#     For Linux s390x, it is linux64-s390x
-#     For Linux x64, it is linux-x86_64
-#     For Mac OSX, it is darwin64-x86_64-cc
-#     For Windows x64, it is VC-WIN64A
-#     For Windows x32, it is VC-WIN32
-build_openssl:
-ifeq ($(BUILD_OPENSSL), yes)
-	$(ECHO) Compiling OpenSSL in $(OPENSSL_DIR)
-  ifeq ($(OPENJDK_TARGET_OS), linux)
-    ifeq ($(OPENJDK_TARGET_CPU), x86_64)
-	($(CD) $(OPENSSL_DIR) && ./Configure linux-x86_64 shared && $(MAKE))
-    else ifeq ($(OPENJDK_TARGET_CPU), s390x)
-	($(CD) $(OPENSSL_DIR) && ./Configure linux64-s390x shared && $(MAKE))
-    else ifeq ($(OPENJDK_TARGET_CPU), ppc64le)
-	($(CD) $(OPENSSL_DIR) && ./Configure linux-ppc64le shared && $(MAKE))
-    else
-	($(CD) $(OPENSSL_DIR) && ./config shared && $(MAKE))
-    endif
-  else ifeq ($(OPENJDK_TARGET_OS), macosx)
-	($(CD) $(OPENSSL_DIR) && ./Configure darwin64-x86_64-cc shared  && $(MAKE))
-  else ifeq ($(OPENJDK_TARGET_OS), windows)
-    ifeq ($(OPENJDK_TARGET_CPU), x86_64)
-	($(CD) $(OPENSSL_DIR) && ./Configure VC-WIN64A shared && $(MAKE))
-    else
-	($(CD) $(OPENSSL_DIR) && ./Configure VC-WIN32 shared && $(MAKE))
-    endif
-  else ifeq ($(OPENJDK_TARGET_OS), aix)
-	($(CD) $(OPENSSL_DIR) && ./Configure aix64-cc shared && $(MAKE))
+ifeq ($(OPENJDK_TARGET_OS), windows)
+  # Special setup required for Windows:
+  # - OpenSSL generates an nmake-style makefile so we must use nmake
+  OPENSSL_MAKE := nmake
+  # - CONFIGURE_INSIST bypasses the requirement that perl use Windows-style paths (with backslashes)
+  # - LIB cannot have the surrounding double-quotes provided by custom-spec.gmk
+  # - PERL must be a Windows-style path that will be usable by nmake
+  # - MAKEFLAGS uses unix-style options (with a dash) which won't be understood by nmake
+  OPENSSL_SETUP := export \
+		CONFIGURE_INSIST=true \
+		LIB='$(subst ",,$(LIB))' \
+		PERL='$(shell $(CYGPATH) -m $(PERL))' \
+	&& unset MAKEFLAGS \
+	&&
+else
+  OPENSSL_MAKE := $(MAKE)
+  OPENSSL_SETUP :=
+endif
+
+# Identify the desired openssl target configuration
+# (the default is to auto-detect).
+OPENSSL_TARGET :=
+ifeq ($(OPENJDK_TARGET_OS), aix)
+  OPENSSL_TARGET := aix64-cc
+else ifeq ($(OPENJDK_TARGET_OS), linux)
+  ifeq ($(OPENJDK_TARGET_CPU), ppc64le)
+    OPENSSL_TARGET := linux-ppc64le
+  else ifeq ($(OPENJDK_TARGET_CPU), s390x)
+    OPENSSL_TARGET := linux64-s390x
+  else ifeq ($(OPENJDK_TARGET_CPU), x86_64)
+    OPENSSL_TARGET := linux-x86_64
+  endif
+else ifeq ($(OPENJDK_TARGET_OS), macosx)
+  ifeq ($(OPENJDK_TARGET_CPU), arm64)
+    OPENSSL_TARGET := darwin64-x86_64-cc
```

Comment:
This specifies x86 instead of arm.

Reply:
Good catch. I inverted that 'ifeq' element badly.

Reply:
I plan to replace the series of CPU cases to:
```
  ifneq (,$(filter arm64 x86_64, $(OPENJDK_TARGET_CPU)))
    OPENSSL_TARGET := darwin64-$(OPENJDK_TARGET_CPU)-cc
  endif
```
and apply a similar change for Linux.